src/ConnectedMode/ServerSentEvents/IServerSentEventSource.cs

```diff
@@ -26,21 +26,22 @@ namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
     /// <summary>
     /// Source for the server sent events about certain server side changes to the SQ/SC project
     /// </summary>
+    /// <remarks>This interface is not intended to be thread safe</remarks>
     /// <typeparam name="T">Server sent event type inherited from <see cref="IServerEvent"/></typeparam>
-    internal interface IServerSentEventSource<T> where T : IServerEvent
+    internal interface IServerSentEventSource<T> where T : class, IServerEvent
```

Comment:
This set of interfaces might need to be implemented in the Core assembly. The implementation of Taint will not be here but in the Taint assembly.